flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,257 @@ import Sentry
 import UIKit
 
 public class SwiftSentryFlutterPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
-    let instance = SwiftSentryFlutterPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
-
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
-  }
-
-  override init() {
-    if let path = Bundle.main.path(forResource: "Info", ofType: "plist") {
-      if let resource = NSDictionary(contentsOfFile: path) {
-        if let dsn = resource.object(forKey: "SentryDsn") {
-          SentrySDK.start { options in
-              options.dsn = dsn as? String
-              options.debug = resource["SentryDebug"] as? Bool ?? false
-              options.attachStacktrace = true
-          }
-        }
-      }
-    }
-  }
+
+    var sentryOptions: Options?
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let channel = FlutterMethodChannel(name: "sentry_flutter", binaryMessenger: registrar.messenger())
+        let instance = SwiftSentryFlutterPlugin()
+        registrar.addMethodCallDelegate(instance, channel: channel)
+    }
+
+    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+        switch call.method as String{
+            case "deviceInfos" :
+              deviceInfos(result: result)
+              break
+
+            case "initNativeSdk" :
+              initNativeSdk(call, result: result)
+              break
+
+        case "captureEnvelope" :
+            captureEnvelope(call, result: result)
+            break
+        default:
+            result(FlutterMethodNotImplemented)
+        }
+
+
+    }
+
+    private func deviceInfos(result: @escaping FlutterResult){
+        SentrySDK.configureScope{ scope in
+            let serializedScope = scope.serialize()
+            let contexts = serializedScope["context"]
+
+            var infos = ["contexts":contexts]
+            print("contexts \(String(describing: contexts))")
+            // TODO DEBUG context
+
+            if let integrations = self.sentryOptions?.integrations {
+                infos["integrations"] = integrations
+            }
+
+            // TODO add sdk.packages
+            result(infos)
+        }
+    }
+
+    private func initNativeSdk(_ call: FlutterMethodCall, result: @escaping FlutterResult){
+        guard let arguments = call.arguments as? [String: Any] else {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
+
+        if (arguments.isEmpty) {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
```

Comment:
set a new code and a specific message as one tests if its null and the other if its empty

Reply:
✅ move the two conditions to the `guard`